TheMagicApprentice/modules/entities/player/spells/sun_beam/SunBeam.cs

```diff
@@ -1,19 +1,206 @@
+	*/
+	private void UpdateMesh(List<Vector2> points)
+	{
+		var arrays = new Godot.Collections.Array();
+		arrays.Resize((int)Mesh.ArrayType.Max);
+
+		var vertices = points.ToArray();
+		var colors = new Color[vertices.Length];
+		var indices = new int[(vertices.Length - 2) * 3];
 
-		LookAt(targetPosition); // make spell look in the correct direction
+		for (int i = 0; i < vertices.Length; i++)
+		{
+			colors[i] = BeamColor;
+		}
+
+		int index = 0;
+		for (int i = 1; i < vertices.Length - 1; i++)
+		{
+			indices[index++] = 0; // The center point (player position)
+			indices[index++] = i;
+			indices[index++] = i + 1;
+		}
+
+		arrays[(int)Mesh.ArrayType.Vertex] = vertices;
+		arrays[(int)Mesh.ArrayType.Color] = colors;
+		arrays[(int)Mesh.ArrayType.Index] = indices;
+
+		/*var arrayMesh = new ArrayMesh();
+		arrayMesh.AddSurfaceFromArrays(Mesh.PrimitiveType.Triangles, arrays);
+
+		meshInstance.Mesh = arrayMesh;*/
+	}
+
+	/**
+	Updates the collision polygon for the beam.
+	@param points The list of points defining the beam's shape
+	*/
+	private void UpdateCollisionPolygon(List<Vector2> points)
+	{
+		// Create a list for the polygon points
+		var polygonPoints = new List<Vector2>();
+		
+		// Add the center point (origin of the beam)
+		polygonPoints.Add(Vector2.Zero);
+		
+		// Add all other points
+		polygonPoints.AddRange(points.GetRange(1, points.Count - 1));
+		
+		// Remove any duplicate points
+		polygonPoints = polygonPoints.Distinct().ToList();
+
+		// Ensure the polygon is convex by using a convex hull algorithm
+		polygonPoints = ConvexHull(polygonPoints);
+
+		collisionPolygon.Polygon = polygonPoints.ToArray();
+	}
+
+	/**
+	Computes the convex hull of a set of points.
+	@param points The list of points to compute the convex hull for
+	@return The list of points forming the convex hull
+	*/
+	private List<Vector2> ConvexHull(List<Vector2> points)
+	{
+		if (points.Count <= 3)
+			return points;
+
+		// Sort points lexicographically
+		points.Sort((a, b) => a.X != b.X ? a.X.CompareTo(b.X) : a.Y.CompareTo(b.Y));
+
+		List<Vector2> hull = new List<Vector2>();
+
+		// Build lower hull
+		foreach (var point in points)
+		{
+			while (hull.Count >= 2 && Cross(hull[hull.Count - 2], hull[hull.Count - 1], point) <= 0)
+				hull.RemoveAt(hull.Count - 1);
+			hull.Add(point);
+		}
+
+		// Build upper hull
+		int lowerHullSize = hull.Count;
+		for (int i = points.Count - 2; i >= 0; i--)
+		{
+			Vector2 point = points[i];
+			while (hull.Count > lowerHullSize && Cross(hull[hull.Count - 2], hull[hull.Count - 1], point) <= 0)
+				hull.RemoveAt(hull.Count - 1);
+			hull.Add(point);
+		}
+
+		hull.RemoveAt(hull.Count - 1);  // Last point is the same as the first one
+
+		return hull;
+	}
+
+	/**
+	Computes the cross product of three points.
+	@param o The origin point
+	@param a The first point
+	@param b The second point
+	@return The cross product value
+	*/
+	private float Cross(Vector2 o, Vector2 a, Vector2 b)
```

Comment:
Die Funktion sollte static sein.
Die Funktion gibt es btw bereits in Godot: (a-o).Cross(b-o)